code/headers/gas_container_c.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+#include <array>
+#include <gas_s.hpp>
+namespace r2d2::gas_detection {
+    /**
+     * Class gas_container_c is a containers for all the gas values.
+     */
+    template <int AmountOfGasses>
+    class gas_container_c {
```

Comment:
If you have a std::array of gasses and each gas is unique, you could use the `gas_id` as the index into the array.
That way you don't need this class at all.

Reply:
Thanks for your response. 
The way this is designed is so that it is easily identified which gasses are measured. It is **not** by default that all gasses will be measured the entire time. For example: The MQ-2 measures LPG,  CO and Smoke(gas ID 0, 1 and 2 respectively). The MQ-3 measures only alcohol(gas ID 3). 

For example: If only the MQ-3 is used, the index wouldn't work with the gas_container class.

Reply:
I see, it's fine then!